mode/soy/soy.js

```diff
@@ -125,6 +125,44 @@
       this.scope = scope;
     }
 
+    function expression(stream, state) {
+      let match;
+      if (stream.match(/[[]/)) {
+        state.soyState.push("list-literal");
+        state.lookupVariables = false;
+        return null;
+      } else if (stream.match(/map/)) {
```

Comment:
I suspect you'll need at least a `\b` at the end of this regular expression, to avoid matching any word that starts with "map". (Same for other word regexps.)

Reply:
Yes, makes sense. Added it to all places I could find.